code/modules/vending/_vending.dm

```diff
@@ -331,7 +331,9 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 				price_listed = "$[R.custom_price]"
 			if(!onstation || account && account.account_job && account.account_job.paycheck_department == payment_department)
 				price_listed = "FREE"
-			if(coin_records.Find(R) || is_hidden)
+				if(coin_records.Find(R))
+					price_listed = "$[extra_price]"
+			else if(coin_records.Find(R) || is_hidden)
```

Comment:
Do you need to check if `coin_records.Find(R)` here? I think only the `is_hidden` is needed

Reply:
coin_records.Find(R) is for premium item, is_hidden is for contraband
The if(coin_records.Find(R)) makes that premium items are still expensive for department while contrabands are free.
The else if(coin_records.Find(R) || is_hidden) is for anyone else outside the derpartment, which both premium items and contrabands will be expensive.

Reply:
I mean, if coin_records.Find(R) is true, it will stay there and execute code. If it is false, then it goes to the elseif, where it checks if its true again, which is not needed, since you already checked it before, right?

Reply:
The else is for the previous if, which is if(!onstation || account && account.account_job && account.account_job.paycheck_department == payment_department) and detects if the person is from the department of the vendor. So the else if will just happen if the person is not from the department of the vendor.

Reply:
Fuck, I'm dumb. sorry. It considers indentation for ifs.

Reply:
Wait, I need to tell I approve it because I commented on it? I'm bad at github

Reply:
No you used to be able to consolidate reviews that are no longer applicable but evil hitler maintainers removed it from contribs.

Reply:
Oh, I see. PR to add "Maintainers" costume to curator that is actually a nazi outfit.